src/vllm_tgis_adapter/grpc/grpc_server.py

```diff
@@ -512,8 +512,17 @@ def _convert_output(  # noqa: PLR0913
         return response
 
     @staticmethod
-    def request_id(context: ServicerContext) -> str:  # noqa:  ARG004
-        return uuid.uuid4().hex
+    def request_id(context: ServicerContext) -> str:
+        metadata = context.invocation_metadata()
+        if not metadata:
+            return uuid.uuid4().hex
+
+        correlation_id = dict(metadata).get("x-correlation-id")
+
+        if not correlation_id:
+            return uuid.uuid4().hex
+
+        return correlation_id
```

Comment:
Is the compact version clearer to read?
```suggestion
         metadata = dict(context.invocation_metadata() or {})
         return metadata.get("x-correlation-id", uuid.uuid4().hex)
```

Reply:
Compared to the compact version, we avoid evaluating `uuid.uuid4()` when we have the correlation id with the current code, I'm fine either way tbh

Reply:
You're right, this can be addressed by replacing the return statement with:
```python
return metadata.get("x-correlation-id") or uuid.uuid4().hex
```
I was initially triggered by the repeated use of `uuid4()`, aiming to follow the DRY principle, but I'm okay with whichever approach you prefer.